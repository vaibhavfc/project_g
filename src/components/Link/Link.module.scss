@import '../../sass/theme/default';

.root {
  //! TODO: font-family is not applying
  @include body-3;
  color: $link-root;
  cursor: pointer;

  &:hover {
    color: $link-root;
  }

  &:active {
    color: $link-root__active;
  }

  &:focus {
    outline: none;
    text-decoration: underline;
    text-underline-offset: 1px;
    text-decoration-thickness: 2px;
  }

  &:disabled {
    color: $link-root__disabled;
    pointer-events: none;
  }

  &:visited {
    color: $purple;
  }
}

.inlineLink {
  text-decoration: underline;
}

.disabled {
  color: $link-root__disabled;
  pointer-events: none;
}

.active {
  color: $link-active !important;
}
