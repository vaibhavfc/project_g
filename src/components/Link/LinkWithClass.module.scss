
// .root {
//   color: #ED1164;
//   cursor: pointer;

//   &:hover {
//     // text-decoration: underline;
//     color: #ED1164;
//   }  
// }

// // .label {
// //   color: pink;
// //   margin-left: 400px;
// // }