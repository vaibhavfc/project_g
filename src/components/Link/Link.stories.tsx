// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import Link from '.';

import classes from './LinkWithClass.module.scss'

import { LinkProps } from './Link.type';

export default {
  title: 'Components/Link',
  component: Link,
  parameters: {
    controls: {
      // disabled: true,
    }
  }
} as Meta;

const Template: Story<LinkProps> = (args) => <Link {...args}></Link>

export const DefaultLink = Template.bind({});
DefaultLink.args = {
  children: 'Default Link',
  to: 'default_link'
}

export const InlineLink = Template.bind({});
InlineLink.args = {
  children: 'Inline Link',
  inlineLink: true,
  to: 'inline_link'
}

export const ActiveLink = Template.bind({})
ActiveLink.args = {
  children: 'Active Link',
  active: true,
  to: 'active_link'
}
