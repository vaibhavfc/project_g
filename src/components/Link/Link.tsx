
import React, { FC, forwardRef, useMemo } from 'react'
import { LinkProps, LinkStylingProps } from './Link.type'
import { Link as MuiLink } from "@material-ui/core"
import { Link as RouterLink } from 'react-router-dom';
import { MemoryRouter as Router } from 'react-router';
import classes from './Link.module.scss'

import classnames from 'classnames';

import { mergeClassesObjects } from '../../helpers/styling/mergeClassesObjects';

const LinkWithInnerRef: FC<LinkProps> = ({
  innerRef = null,
  classes: overrideClasses = {},
  children = 'link',
  disabled = false,
  inlineLink = false,
  tabIndex=0,
  active,
  to="#",
  doneCallback = () => undefined,
  ...props
}) => {
  const mergedClasses = useMemo(
    () => mergeClassesObjects<LinkStylingProps>(classes, overrideClasses),
    [overrideClasses],
  );
  return (
    <Router>
      <MuiLink
        component={RouterLink}
        tabIndex={tabIndex}
        classes={{
          root: classnames(mergedClasses.root, { [classes.inlineLink]: inlineLink, [classes.disabled]: disabled, [classes.active]: active  })
        }}
        to={to}
      >
        {children}
      </MuiLink>
    </Router>)
}

const Link = forwardRef<HTMLAnchorElement, Omit<LinkProps, 'innerRef'>>(
  (props, ref) => <LinkWithInnerRef innerRef={ref} {...props} />
);
export default Link;