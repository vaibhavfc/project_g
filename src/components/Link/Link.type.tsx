import { Ref } from 'react'
import { LinkProps as MuiLinkProps, LinkClassKey } from '@material-ui/core/Link'
export interface LinkStylingProps extends Partial<Record<LinkClassKey, string>> {
}

export interface LinkProps extends MuiLinkProps {
  innerRef: Ref<HTMLAnchorElement>;
  disabled?: boolean;
  inlineLink?: boolean;
  active?:boolean;
  to?:string;
  doneCallback?: () => void;
}
