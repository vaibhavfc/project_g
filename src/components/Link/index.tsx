import Link from './Link';
import { LinkProps } from './Link.type';

export type { LinkProps }

export default Link;