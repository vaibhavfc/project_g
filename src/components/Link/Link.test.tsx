import { render } from '@testing-library/react';

import renderer from 'react-test-renderer';

import Link from './Link';

describe('test', () => {
  it('Renders Link Compoents', () => {
    render(<Link>test</Link>);
  });

  it('Matches Snapshot of Link', () => {
    const tree = renderer.create(<Link> TEST</Link>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
