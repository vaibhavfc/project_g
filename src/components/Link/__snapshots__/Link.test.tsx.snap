// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Matches Snapshot of Link 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  href="/home"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  tabIndex={0}
>
   to workd 
</a>
`;
